### PR TITLE
[MI-2318] Added unit tests for some components

### PR DESCRIPTION
**Pull request by manojmalik20**
Added unit tests for IconButton, Spinner and RHSHeader

### CODE REVIEW

File: webapp/src/components/Buttons/iconButton/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        };
+        const wrapper = shallow(<IconButton {...props}/>);
+        expect(wrapper).toMatchSnapshot();
+
+        expect(wrapper.find('Button').hasClass(props.extraClass)).toBeTruthy();
+        expect(wrapper.find('i').hasClass(props.iconClassName)).toBeTruthy();
+        wrapper.find('Button').simulate('click');
+        expect(props.onClick).toHaveBeenCalled();
+    });
+});
```
Comment (ayusht2810) [inline webapp/src/components/Buttons/iconButton/index.test.tsx:32]: add eof

File: webapp/src/components/Buttons/iconButton/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should match snapshot with all the props', () => {
+        const props = {
+            ...baseProps,
+            iconClassName: 'mockIconClassName',
+            extraClass: 'mockExtraClass',
+            onClick: jest.fn(),
+            children: (<span>{'Dummytext'}</span>),
```
Comment (ayusht2810) [inline webapp/src/components/Buttons/iconButton/index.test.tsx:22]:
```suggestion
            children: <></>,
```

File: webapp/src/components/spinner/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    const baseProps = {
+        extraClass: 'mockClassName',
+    };
+
+    it('should match snapshot with correct className', () => {
+        const wrapper = shallow(<Spinner {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+        expect(wrapper.find('div').hasClass(baseProps.extraClass)).toBeTruthy();
+    });
+});
```
Comment (ayusht2810) [inline webapp/src/components/spinner/index.test.tsx:16]: add eof

File: webapp/src/containers/Rhs/rhsHeader.test.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        wrapper.find('IconButton').simulate('click');
+        expect(wrapper).toMatchSnapshot();
+        expect(wrapper.find('ToggleSwitch').exists()).toBeTruthy();
+
+        wrapper.find('Button').first().simulate('click');
+        expect(baseProps.setShowAllSubscriptions).toBeCalled();
+        expect(baseProps.setFilter).toBeCalled();
+        expect(baseProps.setResetFilter).toBeCalled();
+    });
+});
```
Comment (ayusht2810) [inline webapp/src/containers/Rhs/rhsHeader.test.tsx:44]: add eof

**Review by ayusht2810 — APPROVED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by sooraj-shukla — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**